### ISSUE TITLE
Question: software_* in SSA as defaults for standard client metadata

### ISSUE THREAD

**Issue by TakahikoKawasaki** [OpenBanking-Brasil/specs-seguranca#114]
The example in "[8.1. Software Statement Claims](https://openbanking-brasil.github.io/specs-seguranca/open-banking-brasil-dynamic-client-registration-1_ID1.html#name-software-statement-claims)" of "[Open Banking Brasil Financial-grade API Dynamic Client Registration 1.0 Implementers Draft 1](https://openbanking-brasil.github.io/specs-seguranca/open-banking-brasil-dynamic-client-registration-1_ID1.html) shows that a software statement issued by the Directory contains some claims that start with "`software_`" prefix. Some of them can generate standard claim names by removing the prefix. For example, removing `software_` from `software_client_name` generates `client_name` which is defined in OpenID Connect Dynamic Client Registration 1.0.

Should values of `software_*` claims be used as default values for corresponding standard claims? Or should we avoid doing it?

To be concrete, I'm wondering if values of the claims in "Claim in SSA" column in the table below should be used as default values for standard claims in "Standard Claim" column.

| Claim in SSA | Standard Claim |
|:--|:--|
|`software_client_name`|`client_name`|
|`software_tos_uri`|`tos_uri`|
|`software_client_description`|`client_description`|
|`software_policy_uri`|`policy_uri`|
|`software_client_uri`|`client_uri`|
|`software_logo_uri`|`logo_uri`|

The table above intentionally excludes `software_redirect_uris` and `software_jwks_uri` because OBB DCR has explicit rules for them. In addition, the table excludes `software_client_id` because it is different from `client_id` which will be assigned by the authorization server as a result of the dynamic client registration.

**Comment by RalphBragg** [comment]
@TakahikoKawasaki  - this was raised on the security working group and was passed to the 'User Experience WG'. The reason the requirements for Banks to enforce or use these other fields isn't defined is that they Authorisation Server user experience guidelines haven't been shared (at least not with me).

If the CEG's for Authorization Servers are published that state, Banks MUST show the client URI and client logo of the oAuth 2.0 client then the requirement to display and present those claims will be included on the spec. 

@ginglass @DeiGratia33  - do you have anything back from the CX/UI/UX people on the requirements for Banks to display what information regarding the app?

@TakahikoKawasaki  - there is nothing wrong with defaulting the claims, indeed it should be encouraged provided that putting all of this information on a consent screen doesn't mean that the Bank's AS would be in breach of the CX/UI/UX guidelines which i haven't seen.

**Comment by ginglass** [comment]
@RalphBragg , personally, I don't... 

I thought that `software_` prefix was intentionally made just to differ this claims sent from TPP to directory (with _ROLES_ etc)  of those standard that should to be returned by directory (with the scopes related to regulatory roles and signed by directory). 

I thought this SSA JWT returned by directory should be the one that TPP have to inform on `software_statement` atribute sent to ASPSP `/req`... this one RFC7591 compliant.

Unfortunately I realize that I´m wrong because JWT present on `software_statemente`atribute sent do ASPSP `/req` is the same sent to directory (with roles etc)...

How is the format of JWT returned by directory and that have to be present on software_statement request attibute `/req`?

Where ASPSP will get the relationship bw ROLES and scopes?

Tks

**Comment by RalphBragg** [comment]
I'm not sure I understand the question. There are worked examples that are published on the open ID certification suite - dcr examples that I went though yesterday.

The scope mappings for regulatory roles is included in the table at the bottom of the dcr speciation.

**Comment by ginglass** [comment]
ASPSP have to keep relationship bw cliente_id issued and *scopes*, right? scopes are not present at software_statement JWT nor RFC7591 request shown on section 9.1... so AS have to keep the relationship bw cliente_id and authorized ROLES and scopes... and the relationship bw ROLES and scopes have to be coded inside AS. Is that right?

**Comment by RalphBragg** [comment]
That is correct; the regulatory roles of DADOS defines what a TPP actually needs to have access to. The scopes don't actually add anything because the TPP must be given permission to request all of the scopes. Likewise if a Bank staggers roll out of more endpoints for the APIs they have to make sure the the TPP can access those endpoints if it is a DADOS. How each bank maintains this is up to each bank and different vendors will do it differently. For example, with our banks, we include the consent payload in the introspection and our API Gateway will grant access based on the Consent Permissions because this is actually what the customer has consented too.

**Comment by ginglass** [comment]
Could you please share with us a sample of Response Object ("SSL JWT") sent from DirectoryAPI to TTP on step 12 of DCR doc (Dynamic Client Registration Request Processing)?

**Comment by RalphBragg** [comment]
I'm really confused about the request, I did a record demonstration to the 300+ people about how you can use the mock bank and the dcr certification tests to look at all of the end to end flows and even try it yourself.
https://www.certification.openid.net/log-detail.html?log=rPQ7sq4qP2S8IKM&public=true
https://gitlab.com/openid/conformance-suite/-/wikis/Brazil-DCR-Example-Configuration

**Comment by ginglass** [comment]
My request was just to figure out if we should put this sample on DCR doc to make it more clear... thanks for this refs.

**Comment by RalphBragg** [comment]
No worries, perhaps we should add an appendix to the DCR document? Or update the appendix with this example?

**Comment by michellebandarraChicago** [comment]
Conforme já comentado pelo Ralph, os valores mencionados na tabela "Claim in SSA" (utilizada no exemplo 8.1) podem ser considerados para validação dos valores correspondentes na Standard claims, especificado. 
Alerta: Se alteração for realizada no Diretório Central (adição de “software_”) serão necessários ajustes em todas as IFs com implementação vigente
